src/test/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActorTest.scala

```diff
@@ -233,19 +240,20 @@ class TaskLauncherActorTest extends AkkaUnitTest with Eventually {
 
       val lostInstance = TestInstanceBuilder.newBuilder(f.app.id).addTaskUnreachable(unreachableStrategy = unreachableStrategy).getInstance()
 
-      Mockito.when(instanceTracker.instancesBySpecSync).thenReturn(InstanceTracker.InstancesBySpec.forInstances(Seq(lostInstance, Instance.scheduled(f.app))))
+      instanceTracker.instancesBySpec()(any) returns Future.successful(InstanceTracker.InstancesBySpec.forInstances(Seq(lostInstance, Instance.scheduled(f.app))))
       val captor = ArgumentCaptor.forClass(classOf[InstanceOpFactory.Request])
       // we're only interested in capturing the argument, so return value doesn't matter
       Mockito.when(instanceOpFactory.matchOfferRequest(captor.capture())).thenReturn(f.noMatchResult)
 
       val launcherRef = createLauncherRef(constraintApp.id)
-      launcherRef ! RateLimiter.DelayUpdate(constraintApp.configRef, None)
+      rateLimiterActor.expectMsg(RateLimiterActor.GetDelay(f.app.configRef))
+      rateLimiterActor.reply(RateLimiter.DelayUpdate(f.app.configRef, None))
```

Comment:
This avoids the `TaskLauncherActor` to stash any message which does not work. `TestActorRef` and `stash()/unstashAll()` end up in dead locks!

Reply:
That's super weird!!!

Reply:
Great find :)